Add `Blog.restApiRootURL` Core Data property

crazytonyli opened the pull request:
The property will be used to store self hosted site's REST API endpoint URL, which is typically `<site-url>/wp-json`.

## Regression Notes
1. Potential unintended areas of impact


2. What I did to test those areas of impact (or what existing automated tests I relied on)


3. What automated tests I added (or what prevented me from doing so)

PR submission checklist:

- [ ] I have completed the Regression Notes.
- [ ] I have considered adding unit tests for my changes.
- [ ] I have considered adding accessibility improvements for my changes.
- [ ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

Testing checklist:
- [ ] WordPress.com sites and self-hosted Jetpack sites.
- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] VoiceOver.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] iPhone and iPad. 
- [ ] Multi-tasking: Split view and Slide over. (iPad)

dangermattic commented:
<!--
  0 Errors
  2 Warnings: This PR is larger than 500 lin...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 500 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/wordpress-mobile/WordPress-iOS/milestone/308">25.9</a>. The due date for this milestone has already passed.<br />
Please assign it to a milestone with a later deadline or check whether the release for this milestone has already been finished.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

crazytonyli committed:
Add `Blog.restApiRootURL` Core Data property

The property will be used to store self hosted site's REST API endpoint
URL, which is typically `<site-url>/wp-json`.

mokagio commented on WordPress/Classes/WordPress.xcdatamodeld/WordPress 155.xcdatamodel/contents:
```diff
@@ -0,0 +1,883 @@
+        <attribute name="quickStartTypeValue" optional="YES" attributeType="Integer 16" defaultValueString="0" usesScalarValueType="YES" syncable="YES"/>
+        <attribute name="quotaSpaceAllowed" optional="YES" attributeType="Integer 64" usesScalarValueType="NO" syncable="YES"/>
+        <attribute name="quotaSpaceUsed" optional="YES" attributeType="Integer 64" usesScalarValueType="NO" syncable="YES"/>
+        <attribute name="restApiRootURL" optional="YES" attributeType="String" syncable="YES"/>
```
This is the line in the model with the new property.

mokagio reviewed (approved):
Also diffed 154 with the newest version added in this PR:

```diff
➜ diff WordPress/Classes/WordPress.xcdatamodeld/WordPress\ {154,155}.xcdatamodel/contents
2c2
< <model type="com.apple.IDECoreDataModeler.DataModel" documentVersion="1.0" lastSavedToolsVersion="22757" systemVersion="23E224" minimumToolsVersion="Xcode 9.0" sourceLanguage="Swift" userDefinedModelVersionIdentifier="">
---
> <model type="com.apple.IDECoreDataModeler.DataModel" documentVersion="1.0" lastSavedToolsVersion="23788" systemVersion="24D81" minimumToolsVersion="Xcode 9.0" sourceLanguage="Swift" userDefinedModelVersionIdentifier="">
159a160
>         <attribute name="restApiRootURL" optional="YES" attributeType="String" syncable="YES"/>
```

Looking good.

kean commented on WordPress/Classes/WordPress.xcdatamodeld/.xccurrentversion:
```diff
@@ -3,6 +3,6 @@
 <dict>
 	<key>_XCCurrentVersionName</key>
-	<string>WordPress 154.xcdatamodel</string>
+	<string>WordPress 155.xcdatamodel</string>
```
let's remember to update this pre-production model with new namespace names.
cc @mokagio pining just in case – I think you saw it already.
